From 586363e7aa7dd63e971b82022868254a26b3fbbe Mon Sep 17 00:00:00 2001
From: Abhay Krishna Arunachalam <arnchlm@amazon.com>
Date: Mon, 10 Mar 2025 23:56:00 -0700
Subject: [PATCH 13/14] Update Packer VM hardware version to support RHEL 9

The RHEL 9 Guest OS type was introduced in vSphere API release 7.0.1.0 (Source - https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.GuestOsDescriptor.GuestOsIdentifier.html),
which is compatible with virtual machine hardware version 18 (Source - https://knowledge.broadcom.com/external/article/315655/virtual-machine-hardware-versions.html).
But image-builder hardcodes the VM hardware version to 15, so we need to override it to 18 for RHEL 9 OVA.

Also the OVA build Python script was missing RHEL 9 in the OS ID mapping, so added it there.

Signed-off-by: Abhay Krishna Arunachalam <arnchlm@amazon.com>
---
 images/capi/hack/image-build-ova.py | 1 +
 images/capi/packer/ova/rhel-9.json  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/images/capi/hack/image-build-ova.py b/images/capi/hack/image-build-ova.py
index a1b9eee72..d557bbd09 100755
--- a/images/capi/hack/image-build-ova.py
+++ b/images/capi/hack/image-build-ova.py
@@ -112,6 +112,7 @@ def main():
                  "centos8-64": {"id": "107", "version": "8", "type": "centos8_64Guest"},
                  "rhel7-64": {"id": "80", "version": "7", "type": "rhel7_64Guest"},
                  "rhel8-64": {"id": "80", "version": "8", "type": "rhel8_64Guest"},
+                 "rhel9-64": {"id": "80", "version": "9", "type": "rhel9_64Guest"},
                  "rockylinux-64": {"id": "80", "version": "", "type": "rockylinux_64Guest"},
                  "ubuntu-64": {"id": "94", "version": "", "type": "ubuntu64Guest"},
                  "flatcar-64": {"id": "100", "version": "", "type": "other4xLinux64Guest"},
diff --git a/images/capi/packer/ova/rhel-9.json b/images/capi/packer/ova/rhel-9.json
index a3f7becff..0b806d1a8 100644
--- a/images/capi/packer/ova/rhel-9.json
+++ b/images/capi/packer/ova/rhel-9.json
@@ -16,5 +16,6 @@
   "os_display_name": "RHEL 9",
   "redhat_epel_rpm": "https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm",
   "shutdown_command": "shutdown -P now",
+  "vmx_version": "18",
   "vsphere_guest_os_type": "rhel9_64Guest"
 }
-- 
2.48.1

